File: testdata/payloads/memory_hog/oom.py
# memory_hog: tests Memory Limit Exceeded (MLE / OOM)
chunks = []
try:
    while True:
        # Allocate 10MB chunks rapidly until cgroup kills process
        chunks.append(bytearray(10 * 1024 * 1024))
except MemoryError:
    pass

File: testdata/payloads/memory_hog/oom.cpp
// memory_hog: tests Memory Limit Exceeded (MLE / OOM)
#include <iostream>
#include <vector>
#include <cstring>

int main() {
    std::vector<char*> allocations;
    while (true) {
        // Allocate 10MB blocks and touch pages to ensure physical resident memory allocation
        char* block = new char[10 * 1024 * 1024];
        std::memset(block, 0xAA, 10 * 1024 * 1024);
        allocations.push_back(block);
    }
    return 0;
}
